src/loaders/OmeZarrLoader.ts

```diff
@@ -16,42 +73,13 @@ class OMEZarrLoader implements IVolumeLoader {
     // get top-level metadata for this zarr image
     const allmetadata = await data.attrs.asObject();
     // each entry of multiscales is a multiscale image.
-    // take the first multiscales entry
-    const imageIndex = 0;
+    const imageIndex = imageIndexFromLoadSpec(loadSpec, allmetadata.multiscales);
     const multiscales = allmetadata.multiscales[imageIndex].datasets;
     const axes = allmetadata.multiscales[imageIndex].axes;
 
-    const axisTCZYX = [-1, -1, -1, -1, -1];
-    for (let i = 0; i < axes.length; ++i) {
-      const axis = axes[i];
-      if (axis.name === "t") {
-        axisTCZYX[0] = i;
-      } else if (axis.name === "c") {
-        axisTCZYX[1] = i;
-      } else if (axis.name === "z") {
-        axisTCZYX[2] = i;
-      } else if (axis.name === "y") {
-        axisTCZYX[3] = i;
-      } else if (axis.name === "x") {
-        axisTCZYX[4] = i;
-      } else {
-        console.log("ERROR: UNRECOGNIZED AXIS in zarr: " + axis.name);
-      }
-    }
+    const axisTCZYX = remapAxesToTCZYX(axes);
     // ZYX
-    const spatialAxes: number[] = [];
-    if (axisTCZYX[2] > -1) {
-      spatialAxes.push(axisTCZYX[2]);
-    }
-    if (axisTCZYX[3] > -1) {
-      spatialAxes.push(axisTCZYX[3]);
-    }
-    if (axisTCZYX[4] > -1) {
-      spatialAxes.push(axisTCZYX[4]);
-    }
-    if (spatialAxes.length != 3) {
-      console.log("ERROR: zarr loader expects a z, y, and x axis.");
-    }
+    const spatialAxes: number[] = findSpatialAxesZYX(axisTCZYX);
```

Comment:
Any reason to type this and the function it calls `[number, number, number]`, or would that just create problems?

Reply:
wouldn't hurt.  I think for proper ome-zarr, Z spatial axis is not required, but for the time being this viewer expects it.